crates/polars-plan/src/dsl/function_expr/array.rs

```diff
@@ -56,6 +57,7 @@ impl ArrayFunction {
             #[cfg(feature = "array_count")]
             CountMatches => mapper.with_dtype(IDX_DTYPE),
             Shift => mapper.with_same_dtype(),
+            Explode => mapper.map_to_list_and_array_inner_dtype(),
```

Comment:
The dtype validation can happen here - e.g.

```rust
Explode => mapper.try_map_to_array_inner_dtype()?,
```

Where `try_map_to_array_inner_dtype` returns an error if the dtype isn't `Array`

Reply:
I don't think that's needed. As that check would be required on every `get_dtype`. We already have this

https://github.com/pola-rs/polars/blob/44439d961e9a0a5236175702bdcc0e4fac076f00/crates/polars-plan/src/plans/conversion/type_coercion/functions.rs#L51

So I think it will resolve itself. During simplification of expressions we can then rewrite `Array(Explode)` to `Explode`.

Reply:
I see - then adding `Explode` under `ArrayFunction` should already add input dtype validation

Reply:
@nameexhaustion 
Thanks, I was thinking about this solution, but I prefered one validation to all namespaces operations.

@ritchie46 
It's not going to wok because we don't hit `check_namespace` due to an extra condition in the match - `if options.cast_to_supertypes.is_some()`:
https://github.com/pola-rs/polars/blob/main/crates/polars-plan/src/plans/conversion/type_coercion/mod.rs#L275

In addition, if we convert to `Expr:Explode`, we may not hit that match at all, I need to verify it.

So I propose -
1. Make `check_namespace` hit all `FunctionExpr`.
2. If the `Array(Explode)` -> `Explode` conversion won't hit, we will implement @nameexhaustion 's idea.

What do you think?